src/ln/channel.rs

```diff
@@ -452,31 +459,46 @@ impl Channel {
+				return Err(HandleError{err: $msg, action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: $msg.to_string() }})});
+			}
+		}
+
 		// Check sanity of message fields:
 		if msg.funding_satoshis >= MAX_FUNDING_SATOSHIS {
-			return Err(HandleError{err: "funding value > 2^24", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("funding value > 2^24");
 		}
 		if msg.channel_reserve_satoshis > msg.funding_satoshis {
-			return Err(HandleError{err: "Bogus channel_reserve_satoshis", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("Bogus channel_reserve_satoshis");
 		}
 		if msg.push_msat > (msg.funding_satoshis - msg.channel_reserve_satoshis) * 1000 {
-			return Err(HandleError{err: "push_msat more than highest possible value", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("push_msat larger than funding value");
 		}
 		if msg.dust_limit_satoshis > msg.funding_satoshis {
-			return Err(HandleError{err: "Peer never wants payout outputs?", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("Peer never wants payout outputs?");
+		}
+		if msg.dust_limit_satoshis > msg.channel_reserve_satoshis {
+			return_error_message!("Bogus; channel reserve is less thand dust limit");
 		}
 		if msg.htlc_minimum_msat >= (msg.funding_satoshis - msg.channel_reserve_satoshis) * 1000 {
-			return Err(HandleError{err: "Minimum htlc value is full channel value", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("Miminum htlc value is full channel value");
 		}
-		Channel::check_remote_fee(fee_estimator, msg.feerate_per_kw)?;
+		Channel::check_remote_fee(fee_estimator, msg.feerate_per_kw).map_err(|e|
+			HandleError{err: e.err, action: Some(msgs::ErrorAction::SendErrorMessage{ msg: msgs::ErrorMessage { channel_id: msg.temporary_channel_id, data: e.err.to_string() }})}
+		)?;
+
 		if msg.to_self_delay > MAX_LOCAL_BREAKDOWN_TIMEOUT {
-			return Err(HandleError{err: "They wanted our payments to be delayed by a needlessly long period", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("They wanted our payments to be delayed by a needlessly long period");
 		}
 		if msg.max_accepted_htlcs < 1 {
-			return Err(HandleError{err: "0 max_accpted_htlcs makes for a useless channel", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return_error_message!("0 max_accpted_htlcs makes for a useless channel");
+		}
+		if msg.max_accepted_htlcs > 483 {
+			return_error_message!("max_accpted_htlcs > 483");
 		}
 		if (msg.channel_flags & 254) != 0 {
-			return Err(HandleError{err: "unknown channel_flags", action: Some(msgs::ErrorAction::DisconnectPeer{ msg: None })});
+			return Err(HandleError{err: "Unknown channel flags", action: Some(msgs::ErrorAction::IgnoreError) });
```

Comment:
Hmm, dont we still need to return the ErrorMessage so that the peer knows we've rejected the channel open request?